src/TaskLineRenderer.ts

```diff
@@ -236,29 +283,15 @@ async function renderComponentText(
 /**
  * The CSS class that describes what the component is, e.g. a due date or a priority, and is a value from LayoutClasses.
  */
-function getTaskComponentClass(component: TaskLayoutComponent, task: Task) {
+function getTaskComponentClass(component: TaskLayoutComponent) {
     const componentClassContainer: string[] = [];
 
-    const componentClass = LayoutClasses[component];
-    switch (component) {
-        case 'blockLink':
-        case 'description':
-        case 'priority':
-        case 'recurrenceRule':
-            componentClassContainer.push(componentClass);
-            break;
-        case 'createdDate':
-        case 'dueDate':
-        case 'startDate':
-        case 'scheduledDate':
-        case 'doneDate': {
-            const date = task[component];
-            if (date) {
-                componentClassContainer.push(componentClass);
-            }
-            break;
-        }
+    const fieldLayoutDetail = FieldLayouts[component];
+    const className = fieldLayoutDetail.className;
+    if (className) {
+        componentClassContainer.push(className);
```

Comment:
I've been trying to work out how this is not a change in behaviour....

The `task` used to be passed in, and the component class was only added if the `date` field value was present in the task. I presume, with our new approval test, the behaviour must be unchanged...

But where has that logic moved to?

Reply:
The task there was used only for this code:

```ts
const date = task[component];
            if (date) {
                componentClassContainer.push(componentClass);
            }
```

However if the date (`task[component]`) is absent it will be absent on higher level - the user of `getTaskComponentClass` won't call `getTaskComponentClass` with a date that is absent =)

Reply:
And before there was an ambiguity in what was the `getTaskComponentClass` called towards. Now this data is contained in `FieldLayoutDetail` class and just resolved in `const fieldLayoutDetail = FieldLayouts[component];`

Reply:
OK, fine to leave it as it is, but two comments:

- As the code is now a little more risky, maintenance wise, because there is now a hidden requirement: I suggest adding a comment in the code of `getTaskComponentClass()` noting that the caller is now responsible for only calling this function if the corresponding field is actually present.
- I see that the JSdoc refers to `LayoutClasses` which no longer exists

Reply:
Ok, I modified the jsdocs including both your comments.

Please advised that the call seems to be fail-safe in TS, for instance, if I try to modify the key of LayoutFields:
![Снимок экрана 2023-11-10 в 15 26 29](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/93825870/ac288b81-2a56-4e37-94f5-e160813201be)

Also note this conversation that is related: https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2403#discussion_r1389134789

Reply:
It's OK, I was unclear. I'll add `1.` myself then merge.

Thanks for sterling efforts.

Reply:
> It's OK, I was unclear. I'll add `1.` myself then merge.

Done in 384c6e08b66e5e72dddc27ea065ff81fd6e70417.